cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler/ArgumentsHandlerTest.kt

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.frontends.python.StatementHandler
+
+import de.fraunhofer.aisec.cpg.TranslationResult
+import de.fraunhofer.aisec.cpg.frontends.python.PythonLanguage
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.get
+import de.fraunhofer.aisec.cpg.test.analyze
+import java.nio.file.Path
+import kotlin.collections.component1
+import kotlin.collections.component2
+import kotlin.collections.iterator
+import kotlin.test.Test
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import org.junit.jupiter.api.BeforeAll
+
+class ArgumentsHandlerTest {
+
+    companion object {
+        private lateinit var topLevel: Path
+        private lateinit var result: TranslationResult
+
+        @JvmStatic
+        @BeforeAll
+        fun setup() {
+            topLevel = Path.of("src", "test", "resources", "python")
+            analyzeFile("arguments.py")
+        }
+
+        fun analyzeFile(fileName: String) {
+            result =
+                analyze(listOf(topLevel.resolve(fileName).toFile()), topLevel, true) {
+                    it.registerLanguage<PythonLanguage>()
+                }
+            assertNotNull(result)
+        }
+    }
+
+    @Test
+    fun testPosOnlyArguments() {
+        var func = result.functions["pos_only_and_args"]
```

Comment:
Some nitpicking: can we make constant stuff `val` instead of `var`?

Reply:
good point. Done